pywbem_mock/_wbemconnection_mock.py

```diff
@@ -0,0 +1,1571 @@
+        if repo.qualifiers:
+            if repo.qualifiers:
+                if not self.qualifiers:
+                    self.qualifiers = repo.qualifiers
+                else:
+                    for ns in repo.qualifiers:
+                        for ql in repo.qualifiers[ns]:
+                            # Test for existing qualifier
+                            self.qualifiers[ns][ql.name] = \
+                                repo.qualifiers[ns][ql.name]
+
+    def add_cimobjects(self, objects, namespace=None):
+        """
+            Add  a CIM object (CIMClass, CIMInstance, CIMQualifierDecl)
+            or list of objects to the faked repository.
+
+            This allows the user to create objects directly and insert them
+            into the repository for tests.
+        """
+        if not namespace:
+            namespace = DEFAULT_NAMESPACE
+        if isinstance(objects, list):
+            for obj in objects:
+                self.add_cimobjects(obj, namespace=namespace)
+
+            if self.verbose:
+                self.display_repository()
+
+        else:
+            obj = objects
+            if isinstance(obj, CIMClass):
+                cc = obj
+                if cc.superclass:
+                    try:
+                        _ = self.GetClass(cc.superclass,  # noqa: F841
+                                          namespace=namespace,
+                                          LocalOnly=True,
+                                          IncludeQualifiers=False)
+                    except CIMError as ce:
+                        if ce.args[0] == CIM_ERR_NOT_FOUND:
+                            ce.args = (CIM_ERR_INVALID_SUPERCLASS,
+                                       cc.superclass)
+                            raise
+                        else:
+                            raise
+                try:
+                    # The following generates an exception for each new ns
+                    self.classes[namespace][cc.classname] = cc
+                except KeyError:
+                    self.classes[namespace] = NocaseDict({cc.classname: cc})
+
+            elif isinstance(obj, CIMInstance):
+                inst = obj
+                if not inst.path:
+                    raise ValueError("Instances added must include path")
+                if not inst.path.namespace:
+                    inst.path.namespace = namespace
+                try:
+                    self.instances[namespace].append(inst)
+                except KeyError:
+                    self.instances[namespace] = [inst]
+
+            elif isinstance(obj, CIMQualifierDeclaration):
+                qual = obj
+                try:
+                    self.qualifiers[namespace][qual.name] = qual
+                except KeyError:
+                    self.qualifiers[namespace] = NocaseDict({qual.name: qual})
+
+            else:
+                raise ValueError('add_cimobjects. %s not valid' % obj)
```

Comment:
In this case, it is an invalid type, so I'd raise TypeError. Plus, the message should point out that the type is not valid. Plus, the type name should appear in the message and not `%s` (which in our case can be things pretty unrelated to the provided object type such as the WBEM URI string for a `CIMInstanceName` object).

Reply:
Done